napari/_tests/test_advanced.py

```diff
@@ -155,6 +155,11 @@ def test_update_console(viewer_factory):
     """Test updating the console with local variables."""
     view, viewer = viewer_factory()
 
+    # Create console
+    assert view.console is None
+    view.toggle_console(None)
+    assert view.console is not None
+
     # Check viewer in console
     assert view.console.kernel_client is not None
```

Comment:
I think this test points to something a bit broken right now: attempting `view.console.whatever` right now will do bad things if it hasn't been instantiated yet. How about making `.console` a property and have it instantiate then if not already created?

Reply:
done, though please check my implementation and its consequences in `toggle_console_visibility`